### ISSUE TITLE
Use TypeScript

### ISSUE THREAD

**Issue by davidbrochart** [jupyter-widgets/ipyleaflet#715]
As the preferred choice for ipywidgets' language in the front-end is TypeScript, should we make the transition in ipyleaflet?

**Comment by martinRenou** [comment]
Yes it would be super nice. Not sure it's a good idea for the sprint though, as it touches the entire code we'll have lots of conflicts.
